Calling proxy.dispose() on proxy using TCPTransport crashes app

### Bug Report
When the proxy is disposed through normal operation using the TCP transport, a NetworkOnMainThread expection is raised and fails to be caught here:

https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android/src/main/java/com/smartdevicelink/transport/TCPTransport.java#L126

This causes the app to crash.

##### Reproduction Steps
1. Have a SDL-integrated Android app connect to Manticore instance with IP addr and port number in its SDL service.
2. Stop the app's SDL service programmtically (calling `proxy.dispose()` as done in https://github.com/smartdevicelink/hello_sdl_android/blob/master/hello_sdl_android/app/src/main/java/com/sdl/hellosdlandroid/SdlService.java#L252)

##### Expected Behavior
While the TCP transport is currently only intended to be used for development purposes, it should not lead to an app crashing on disconnect. 

##### Observed Behavior
Currently an app that uses TCP transport will disconnect if it stops its SDL Service. Here's a stacktrace:

```
java.lang.RuntimeException: Unable to stop service com.smartdevicelink.rpcbuilder.SmartDeviceLink.SdlService@4397a7a: android.os.NetworkOnMainThreadException
at android.app.ActivityThread.handleStopService(ActivityThread.java:3522)
at android.app.ActivityThread.-wrap26(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1697)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6494)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

Caused by: android.os.NetworkOnMainThreadException
at android.os.StrictMode$AndroidBlockGuardPolicy.onNetwork(StrictMode.java:1450)
at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:108)
at java.net.SocketOutputStream.write(SocketOutputStream.java:153)
at com.smartdevicelink.transport.TCPTransport.sendBytesOverTransport(TCPTransport.java:129)
at com.smartdevicelink.transport.SdlTransport.sendBytes(SdlTransport.java:66)
at com.smartdevicelink.SdlConnection.SdlConnection.onProtocolMessageBytesToSend(SdlConnection.java:249)
at com.smartdevicelink.protocol.AbstractProtocol.handlePacketToSend(AbstractProtocol.java:100)
at com.smartdevicelink.protocol.WiProProtocol.EndProtocolSession(WiProProtocol.java:164)
at com.smartdevicelink.SdlConnection.SdlConnection.closeConnection(SdlConnection.java:151)
at com.smartdevicelink.SdlConnection.SdlConnection.unregisterSession(SdlConnection.java:331)
at com.smartdevicelink.SdlConnection.SdlSession.close(SdlSession.java:146)
at com.smartdevicelink.proxy.SdlProxyBase.cleanProxy(SdlProxyBase.java:1422)
at com.smartdevicelink.proxy.SdlProxyBase.dispose(SdlProxyBase.java:1444)
at com.smartdevicelink.rpcbuilder.SmartDeviceLink.SdlService.onDestroy(SdlService.java:329)
```

##### OS & Version Information
* Android Version: 8.1.0
* SDL Android Version: 4.4.0
* Testing Against: Google Pixel 2 XL

##### Test Case, Sample Code, and / or Example App
If you run this app, connect to Manticore instance via TCP and disconnect, the app crashes.
https://github.com/smartdevicelink/rpc_builder_app_android/releases/tag/1.3.0

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/TCPTransport.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/TCPTransport.java
@@ -6,6 +6,9 @@ import java.io.OutputStream;
 import java.net.InetSocketAddress;
 import java.net.Socket;
 
+import android.annotation.TargetApi;
+import android.os.Build;
+import android.os.NetworkOnMainThreadException;
 import android.util.Log;
 
 import com.smartdevicelink.exception.SdlException;
@@ -107,6 +110,7 @@ public class TCPTransport extends SdlTransport {
      * @param length Number of bytes to send
      * @return True if data was sent successfully, False otherwise
      */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
     @Override
     protected boolean sendBytesOverTransport(SdlPacket packet) {
         TCPTransportState currentState = getCurrentState();
@@ -123,7 +127,7 @@ public class TCPTransport extends SdlTransport {
                         mOutputStream.write(msgBytes, 0, msgBytes.length);
                         bResult = true;
                         logInfo("TCPTransport.sendBytesOverTransport: successfully send data");
-                    } catch (IOException e) {
+                    } catch (IOException | NetworkOnMainThreadException e) {
                         logError("TCPTransport.sendBytesOverTransport: error during sending data: " + e.getMessage());
                         bResult = false;
                     }

